appmap/_implementation/event.py

```diff
@@ -210,13 +211,49 @@ def set_params(params, instance, args, kwargs):
             ret.append(p.to_dict(value))
         return ret
 
-    def __init__(self, defined_class, method_id, path, lineno,
-                 fntype, parameters, labels):
+    @property
+    @lru_cache(maxsize=None)
+    def function_name(self):
+        return split_function_name(self._fn)
+
+    @property
+    @lru_cache(maxsize=None)
+    def defined_class(self):
+        return self.function_name[0]
+
+    @property
+    @lru_cache(maxsize=None)
+    def method_id(self):
+        return self.function_name[1]
+
+    @property
+    @lru_cache(maxsize=None)
+    def function_location(self):
+        return get_function_location(self._fn)
+
+    @property
+    @lru_cache(maxsize=None)
+    def path(self):
+        return self.function_location[0]
+
+    @property
+    @lru_cache(maxsize=None)
+    def lineno(self):
+        return self.function_location[1]
+
+    @property
+    @lru_cache(maxsize=None)
+    def comment(self):
+        comment = inspect.getdoc(self._fn)
+        if comment is not None:
+            comment = '\n'.join([f'# {s}' for s in comment.splitlines()])
```

Comment:
I don't think it makes sense to prepend the `#`. The spec certainly doesn't require it (and if it did, it would be really weird with languages that use other comment markers.) I'd just leave the doc as is.

Reply:
The spec doesn't require it, but appmap-swagger does: https://github.com/applandinc/appmap-swagger/blob/main/method.js#L82 . And, the comments emitted by appmap-ruby have the leading `# `.

Reply:
This seems like a problem in appmap-swagger. Can we add a comment here to signal the workaround and open an issue there? It's going to be a problem with other languages.

(Appmap-ruby works like that because it just cuts the code verbatim (sans indentation). It doesn't add the hashes :) But actually it would make sense to remove them universally, once appmap-swagger can handle that.)

Reply:
Yes, I agree that it's a problem in appmap-swagger. I [fixed it](https://github.com/applandinc/appmap-swagger/pull/7), so I'll update this.

Reply:
I think you forgot to remove this after all.